src/api/api.ts

```diff
@@ -152,3 +154,45 @@ function dataURItoFile(dataURI: string) {
   const blob = new Blob([buffer], { type: mimeString });
   return blob;
 }
+
+//Program Enrollment
+export function getPatientEnrolledPrograms(patientUuid: string) {
+  return openmrsFetch(
+    `${BASE_WS_API_URL}programenrollment?patient=${patientUuid}&v=custom:(uuid,display,program,dateEnrolled,dateCompleted,location:(uuid,display))`,
+  ).then(({ data }) => {
+    if (data) {
+      return data;
+    }
+    return null;
+  });
+}
+
+export function createProgramEnrollment(payload, abortController) {
+  if (!payload) {
+    return null;
```

Comment:
@kajambiya I think here you should throw an exception instead of returning null... you should return null when an object is not found (although you can also throw a NFE). Am saying this because the consumer of this method is expecting the created `programEnrolment` but since in this case the object will be null it will look like the enrolment was not created because God knows why but the real cause here is that he passed a null or undefined payload

Reply:
Nice one